pkg/sql/sqlstats/persistedsqlstats/sqlstatsutil/json_impl.go

```diff
@@ -107,11 +107,13 @@ func (s *stmtStatsMetadata) jsonFields() jsonFields {
 type aggregatedMetadata appstatspb.AggregatedStatementMetadata
 
 func (s *aggregatedMetadata) jsonFields() jsonFields {
+	fullScan := s.FullScanCount > 0
 	return jsonFields{
 		{"db", (*stringArray)(&s.Databases)},
 		{"appNames", (*stringArray)(&s.AppNames)},
 		{"distSQLCount", (*jsonInt)(&s.DistSQLCount)},
 		{"failedCount", (*jsonInt)(&s.FailedCount)},
+		{"fullScan", (*jsonBool)(&fullScan)},
```

Comment:
I think if I understand the issue correctly, this is occurring because we're now aggregating the metadata in the server side api. Previously we did that in the frontend so this means that the returned `metadata` column changed from `stmtStatsMetadata` to `aggregatedMetadata`. It looks like we missed that in the original implementation and so we are now trying to decode the aggregated metadata as the single metadata object. I think the fix is not to add these fields to the aggregatedMetadata, but in the combined statements api we should decode the column as the aggregated version and then use that to populate the fields previously expected. WDYT?